### ISSUE TITLE
.robocop not picked up when using API

### ISSUE THREAD

**Issue by fabioz** [MarketSquare/robotframework-robocop#302]
When using the API as:

```
    import robocop
    from robocop.config import Config
    from robocop.utils import issues_to_lsp_diagnostic

    config = Config(root=project_root)
    robocop_runner = robocop.Robocop(config=config)
    robocop_runner.reload_config()

    issues = robocop_runner.run_check(ast_model, filename, source)
    diag_issues = issues_to_lsp_diagnostic(issues)
    return diag_issues
```

The `.robocop` file is not being loaded based on the `project_root`.

I debugged a bit and it seems like `load_default_config_file` is never called. I tried to manually call: `config.parse_opts([])`, right after creating the `Config` and it seems to pick things up on the debugger, but I still couldn't get it to ignore what I expected.

I tested with a file with:

```
*** Test Cases ***
Test
    Fail
```

and a `.robocop` file with:

```
--exclude missing-doc-testcase
--include missing-doc-suite
```

But the missing doc in the testcase is still reported.

So, I'm waiting on some pointers on what I'm doing wrong here...

**Comment by bhirsz** [comment]
I see that ``load_default_config_file`` should be run by ``parse_opts``. parse_opts is only triggered for cli runs. I will update it so parse_opts is always called and release new version (together with fixes for other stuff that was discovered in 1.6.0)

**Comment by fabioz** [comment]
> I see that `load_default_config_file` should be run by `parse_opts`. parse_opts is only triggered for cli runs. I will update it so parse_opts is always called and release new version (together with fixes for other stuff that was discovered in 1.6.0)

Note that just calling it didn't really make it work for my use case.

i.e.: in https://github.com/robocorp/robotframework-lsp/commit/c050f43952baad87819793b6f6d5bc3336063a82, my test is still failing because it's showing the `missing-doc-testcase` when it should be excluded.

Also, note that it shouldn't pick up `sys.argv`, rather, I need to pass the arguments (possibly from the language server configuration).

**Comment by fabioz** [comment]
Also, there's another issue in which calling `parse_opts([])` without a `.robocorp` file present will not work because `load_default_config_file` will return `None` (which will end up calling `sys.exit`).

**Comment by bhirsz** [comment]
You're right, I was able to reproduce it locally. I will work on it.

**Comment by bhirsz** [comment]
Configuration (when calling parse_opts and ``.robocop`` existing) is loaded correctly. The problem is that we're 'filtering out' some if the issues - based on config - when reporting the issues. Since we're not reporting the issues in self.run method it will leave everything. I will rearrange our architecture.

Also if the root is not the Path it will throw an error (I saw you made workaround for that). I will fix that as well.

**Comment by bhirsz** [comment]
> Also, there's another issue in which calling `parse_opts([])` without a `.robocorp` file present will not work because `load_default_config_file` will return `None` (which will end up calling `sys.exit`).

Fixed everything else but I was not being able to reproduce that one. You mean that if there is no `.robocop` file present, running run_check() will stop Python script?

**Comment by fabioz** [comment]
Yes, because it'll pick up the `sys.argv`, which in my case are the arguments to start the language server... if you set arguments that are invalid in `sys.argv` you should also see the error.

Note: ideally an exception would be thrown with the error instead of calling `sys.exit` if parsing the arguments is not ok in the API case...

**Comment by bhirsz** [comment]
I've updated the code so the sys.argv will not be used if you're running robocop using the API. 
I've also added the exception for argument parsing. 
Now I need to add some tests ;) The changes will be released in 1.6.1 later today.

**Comment by bhirsz** [comment]
For argument parsing issues I had to create workaround - because argparse that we're using in Robocp always returns SystemExit. After my changes, if you're using Robocop API and you have invalid configuration it should return "InvalidArgumentError" exception (found in robocop.exceptions).
